module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -171,6 +175,16 @@ class KohaRest extends \VuFind\ILS\Driver\AbstractBase implements
         'Patron::GoneNoAddress' => 'patron_status_address_missing',
     ];
 
+    /**
+     * Item status mappings
+     *
+     * @var array
+     */
+    protected $itemStatusMappings = [
+        // NotForLoan with status number 1 is available in Koha by default:
+        'Item::NotForLoan1' => 'On Reference Desk'
```

Comment:
Just one question: does this status get translated somewhere? Should it? Do we need to add it to the language files somewhere?

Reply:
It can be translated as is, but there's no default translation. SierraRest uses the same string, so I took it from there. I'm not sure if it's a really good string, though, since "Not For Loan" could be something else too. Maybe it would be better to leave it only as an example in the ini file and not map it to anything by default.

Reply:
Then again, the same text is used for ItemType::NotForLoan in getItemStatusCodes.

Reply:
Is "On Reference Desk" the text used in Koha? If not, maybe a more generic "Not Available for Loan" would be less specific (but could still be translated to something else as needed). If so, it probably makes sense for VuFind and Koha's language to match up.

Reply:
There are no hardcoded not for loan status descriptions in Koha at all. It's all fully configurable, with optional defaults: -1:"Ordered", 1:"Not For Loan" and 2:"Staff Collection". (yes, the value could be negative integer)

Reply:
Maybe it's better to not have any default mappings. Then the description from Koha will be displayed, whatever it might be. It can still be translated if necessary.

Reply:
Sounds reasonable to me. Have you tested whether the negative identifier case works reasonably? It might be worth including an example for that just to clarify what the keys would look like to customize those strings. Regarding translation, I can't recall whether we translate status strings at present (though my feeling is that we do not); I wonder if we need to revisit that at a VuFind level, or if it's better to implement translation at the driver level. If you need me to review all the details of current behavior, I'm happy to do so once time permits.

Reply:
Latest version should do it all, and I merged the ItemType status handling with Item status handling. While ItemType doesn't have all the same statuses, the NotForLoan status is pretty similar. Negative values work as well, and I added an example to KohaRest.ini.